code/modules/mining/mining_encounter_objects.dm

```diff
@@ -0,0 +1,164 @@
+			pedestal_2.connected_door = src
+			src.pedestals += pedestal_2
+			var/obj/flock_encounter_pedestal/pedestal_3 = locate(/obj/flock_encounter_pedestal) in pedestal_3_loc
+			pedestal_3.connected_door = src
+			src.pedestals += pedestal_3
+			var/obj/flock_encounter_pedestal/pedestal_4 = locate(/obj/flock_encounter_pedestal) in pedestal_4_loc
+			pedestal_4.connected_door = src
+			src.pedestals += pedestal_4
+
+			pedestal_1.adjacent_pedestals = list(pedestal_2, pedestal_4)
+			pedestal_2.adjacent_pedestals = list(pedestal_1, pedestal_3)
+			pedestal_3.adjacent_pedestals = list(pedestal_2, pedestal_4)
+			pedestal_4.adjacent_pedestals = list(pedestal_1, pedestal_3)
+
+			shuffle_list(src.pedestals)
+			for (var/i in 1 to 2)
+				var/obj/flock_encounter_pedestal/pedestal = src.pedestals[i]
+				pedestal.toggle(null, TRUE)
+
+	disposing()
+		..()
+		src.pedestals = null
+
+	proc/check_puzzle_solved()
+		var/pedestals_on = 0
+		SPAWN(0.1 SECOND) // to prevent race condition for opening the door upon making an activation and deactivation at the same time
+			for (var/obj/flock_encounter_pedestal/pedestal as anything in src.pedestals)
+				if (pedestal.on)
+					pedestals_on++
+			if (pedestals_on >= 4 && src.density)
+				src.open()
+			else if (pedestals_on < 4 && !src.density)
+				src.close()
+
+/obj/item/flock_converter
+	name = "strange tool"
+	desc = "This seems to be a tool of combined Flock, and somehow human, origin, but used for what purpose, you don't know."
+	icon = 'icons/misc/featherzone.dmi'
+	icon_state = "flock_converter"
+	var/contained_mats = 50
+
+	New()
+		..()
+		src.create_inventory_counter()
+		src.inventory_counter.update_number(src.contained_mats)
+
+	attackby(obj/item/W, mob/user)
+		..()
+		if (istype(W, /obj/item/rcd_ammo))
+			playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+			var/obj/item/rcd_ammo/ammo = W
+			src.contained_mats += ammo.matter
+			qdel(ammo)
+			src.inventory_counter.update_number(src.contained_mats)
+
+	afterattack(atom/target, mob/user)
+		..()
+		if (!isturf(target))
+			target = get_turf(target)
+		if (!(istype(target, /turf/simulated) || istype(target, /turf/space)) || isfeathertile(target) || !flockTurfAllowed(target))
+			return
+		if (src.contained_mats < 5)
+			return
+		actions.start(new/datum/action/bar/flock_rcd(target, src), user)
+
+/datum/action/bar/flock_rcd
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_STUNNED | INTERRUPT_ATTACKED
+	duration = 5 SECONDS
+	resumable = FALSE
+
+	var/turf/simulated/target
+	var/obj/item/flock_converter/converter
+	var/obj/decal/decal
+
+	New(turf/simulated/target, obj/item/flock_converter/converter)
+		..()
+		src.target = target
+		src.converter = converter
+
+	onUpdate()
+		..()
+		var/mob/living/critter/flock/F = owner
+		if (!F || isdead(F) || !target || !in_interact_range(F, target) || isfeathertile(target) || !F.can_afford(FLOCK_CONVERT_COST) || !flockTurfAllowed(target))
```

Comment:
You can't just copy paste this if it's meant to be used by a human
![image](https://github.com/user-attachments/assets/0a87c05f-37f4-4f88-b9b7-f0110a9d14a5)